### DISCUSSION TITLE
rust-course/所有权与堆栈讲解的两个问题

### DISCUSSION THREAD

**Discussion by atopx** [sunface/rust-course#303] [内容建议]
# 第`2.3.1`小节中的`所有权与堆栈`
> 当前原文：
![image](https://user-images.githubusercontent.com/58352247/150737667-3dafae8c-9a4b-4944-a107-39b761fed4aa.png)

# 建议：
1. 由于**所有权**一个`rust`的独特核心基础概念(个人理解)，建议增加一些与其他语言(go/java等)之间的对比，让读者更了解为什么`rust`会这样设计，而不是使用其他方法。
2. 期望对这些描述画一些的原理图之类的，辅助读者理解

**Comment by NaturelLee** [comment]
![image](https://user-images.githubusercontent.com/12494855/159201015-9a1e8ff6-e1a1-4611-8512-84d026b9f714.png)

这里堆、栈内存和堆、栈数据结构放在一块儿讲不合适。考虑堆、栈内存和数据结构分开。两部分添加各自的图加以说明。

![image](https://user-images.githubusercontent.com/12494855/159201457-40aaa992-48b1-4886-8228-1ece6395fdc8.png)

数据结构： 意味着结构化数据。
可改为：堆内存上分配的是动态大小的数据。

**Reply by sunface** [reply]
堆和栈是经典的数据结构 :) 这一句话应该没问题的。

然后最开始的建议我觉得较为合理，所有权这块儿内容确实应该再优化下，感谢建议

**Comment by NaturelLee** [comment]
堆栈： 一是堆内存、栈内存；二是堆、栈数据结构；而Rust中说堆栈其实更多的是在讲数据所在的内存是堆内存还是栈内存。

比如：
![image](https://user-images.githubusercontent.com/12494855/159395299-61ddfddd-044c-4e18-a0d6-00390edcef1f.png)
